src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -191,197 +363,138 @@ private ExecutorInfo createExecutor(String taskIdName, String nodeName, String e
-        // Decline any remaining offer
-        for (OfferID offerID : pendingOffers.keySet()) {
-          driver.declineOffer(offerID);
-        }
-        pendingOffers.clear();
-      }
-      return;
-    }
+  private void sendMessageTo(SchedulerDriver driver, TaskID taskId, SlaveID slaveID, String message) {
+    log.info(String.format("Sending message '%s' to taskId=%s, slaveId=%s", message,
+        taskId.getValue(), slaveID.getValue()));
+    String postfix = taskId.getValue();
+    postfix = postfix.substring(postfix.indexOf(".") + 1, postfix.length());
+    postfix = postfix.substring(postfix.indexOf(".") + 1, postfix.length());
+    driver.sendFrameworkMessage(
+        ExecutorID.newBuilder().setValue("executor." + postfix).build(),
+        slaveID,
+        message.getBytes());
+  }
 
-    List<Offer> remainingOffers = new ArrayList<>();
-    remainingOffers.addAll(offers);
+  private boolean isTerminalState(TaskStatus taskStatus) {
+    return (taskStatus.getState().equals(TaskState.TASK_FAILED)
+        || taskStatus.getState().equals(TaskState.TASK_FINISHED)
+        || taskStatus.getState().equals(TaskState.TASK_KILLED)
+        || taskStatus.getState().equals(TaskState.TASK_LOST));
+  }
 
-    if (initializingCluster) {
-      log.info(String.format("Declining remaining %d offers pending initialization",
-          remainingOffers.size()));
-      for (Offer offer : remainingOffers) {
-        driver.declineOffer(offer.getId());
-      }
-      return;
-    }
+  private boolean isRunningState(TaskStatus taskStatus) {
+    return (taskStatus.getState().equals(TaskState.TASK_RUNNING));
+  }
 
-    offers = remainingOffers;
-    remainingOffers = new ArrayList<>();
+  private boolean isStagingState(TaskStatus taskStatus) {
+    return (taskStatus.getState().equals(TaskState.TASK_STAGING));
+  }
 
-    // Check to see if we can launch some DataNodes
-    for (Offer offer : offers) {
-      if (liveState.notInDfsHosts(offer.getSlaveId().getValue())) {
-        launchDataNode(driver, offer);
-      } else {
-        remainingOffers.add(offer);
-      }
+  private void reloadConfigsOnAllRunningTasks(SchedulerDriver driver) {
+    for (Protos.TaskStatus taskStatus : liveState.getRunningTasks().values()) {
+      sendMessageTo(driver, taskStatus.getTaskId(), taskStatus.getSlaveId(),
+          HDFSConstants.RELOAD_CONFIG);
     }
+  }
 
-    // Decline remaining offers
-    log.info(String.format("Declining %d offers", remainingOffers.size()));
-    for (Offer offer : remainingOffers) {
-      driver.declineOffer(offer.getId());
+  private void blockUntilNameNode1Healthy() {
+    //TODO need to implement this method
+    try {
+      Thread.sleep(30000L);
+    } catch (InterruptedException e) {
+      throw new RuntimeException(e);
     }
   }
 
-  @Override
-  public void slaveLost(SchedulerDriver driver, SlaveID slaveId) {
-    log.info("Slave lost slaveId=" + slaveId.getValue());
+  private void correctCurrentPhase() {
```

Comment:
Could the state machine be in a separate class to avoid having the scheduler get too big and to make our code more object oriented?